.github/workflows/validateChangedFiles.sh

```diff
@@ -0,0 +1,35 @@
+#!/bin/bash
+
+INVALID_FILE_PATHS=('pkg/plugin/sdk') # Array of filepaths that are not allowed
+VALID_FILE_PATHS=('pkg/plugin/sdk/v1alpha2') # Array of filepaths that are allowed
+
+# CURRENT_BRANCH_NAME=$(git rev-parse --abbrev-ref HEAD)
+git fetch
+
+CHANGED_FILES=$(git diff origin/main --name-only)
+
+# loop through changed files
+while read file; do
+    fileIsInvalid=0 # 0 = false, 1 = true
+    fileIsValid=0 # 0 = false, 1 = true
+
+    # check if filepath matches an invalid path
+    for invalidPath in "${INVALID_FILE_PATHS[@]}"; do
+        if [[ "${file}" == "${invalidPath}"* ]]; then
+            fileIsInvalid=1
+        fi
+    done
+
```

Comment:
You can simplify all this by just continuing the loop in the case it's whitelisted. If it's not whitelisted and matches a blacklisted, then throw the error. No need for the two variables.